services/ui-src/src/utils/autosave/autosave.ts

```diff
@@ -26,35 +25,50 @@ interface Props {
     userName: string | undefined;
     state: string | undefined;
   };
+  fieldType?: string;
 }
 
 export const autosaveFieldData = async ({
   form,
   fields,
   report,
   user,
+  fieldType,
 }: Props) => {
   const { id, updateReport } = report;
   const { userName, state } = user;
+  let onlyChangedFields = [];
+  switch (fieldType) {
+    case "dynamic":
+      onlyChangedFields = fields.filter((field: FieldInfo) => {
+        const checkedValues = field.displayValues?.filter((el) => el.name);
+        return (
+          checkedValues?.length !== 0 &&
+          field.displayValues !== field?.hydrationValue
+        );
+      });
+      break;
 
-  // if field value hasn't changed from database value, don't autosave field
-  const onlyChangedFields = fields.filter(
-    // TODO: do we need a deeper equality check here?
-    (field: FieldInfo) => {
-      const { value, hydrationValue, defaultValue } = field;
-      return value !== defaultValue && !isEqual(value, hydrationValue);
-    }
-  );
+    default:
+      // if field value hasn't changed from database value, don't autosave field
+      onlyChangedFields = fields.filter(
+        // TODO: do we need a deeper equality check here?
```

Comment:
Note: Good call that we don't need deep equality checking here. From doing a little testing I think the baseline equality checking does just fine for us. 👍